substrate/frame/revive/src/lib.rs

```diff
@@ -1378,17 +1380,17 @@ where
 						data: data.to_vec(),
 					}
 					.into();
-				(result, dispatch_call.get_dispatch_info())
+				(result, dispatch_call)
 			},
 		};
 
 		let Ok(unsigned_tx) = tx.clone().try_into_unsigned() else {
 			return Err(EthTransactError::Message("Invalid transaction".into()));
 		};
 
-		let eth_dispatch_call =
+		let eth_transact_call =
 			crate::Call::<T>::eth_transact { payload: unsigned_tx.dummy_signed_payload() };
-		let fee = tx_fee(eth_dispatch_call, dispatch_info);
+		let fee = tx_fee(eth_transact_call.into(), dispatch_call);
```

Comment:
Maybe we don't even need both arguments. I don't see why we use different calls for the extension weight and the dispatch info.

```suggestion
		let fee = tx_fee(dispatch_call, dispatch_call.get_dispatch_info());
```

Reply:
you need the eth_transact call to get the proper encoded size, the other one give you the actual dispatch_info

Reply:
ahh yes you are right